### PR TITLE
Migrate diff, exec, export, and history to ginkgo

### PR DESCRIPTION
**Pull request by baude**
Migrate the diff, exec, export, and history bats tests to
the ginkgo test suite.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: test/e2e/diff_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(125))
+
+	})
+
+	It("podman rmi with fq name", func() {
+		session := podmanTest.Podman([]string{"rmi", image1})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
```
Comment (mheon) [inline test/e2e/diff_test.go]: Nit: extra whitespace

File: test/e2e/diff_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	})
+
+	It("podman rmi with short name", func() {
+		session := podmanTest.Podman([]string{"rmi", "alpine"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
```
Comment (mheon) [inline test/e2e/diff_test.go]: Nit: whitespace

File: test/e2e/diff_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	})
+
+	It("podman rmi all images", func() {
+		podmanTest.PullImages([]string{image3})
+		session := podmanTest.Podman([]string{"rmi", "-a"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
```
Comment (mheon) [inline test/e2e/diff_test.go]: Nit: empty line

File: test/e2e/diff_test.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	})
+
+	It("podman rmi all images forceably with short options", func() {
+		podmanTest.PullImages([]string{image3})
+		session := podmanTest.Podman([]string{"rmi", "-fa"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
```
Comment (mheon) [inline test/e2e/diff_test.go]: Nit: empty line

File: test/e2e/exec_test.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		if err != nil {
+			os.Exit(1)
+		}
+		podmanTest = PodmanCreate(tempdir)
+		podmanTest.RestoreAllArtifacts()
+	})
+
+	AfterEach(func() {
+		podmanTest.Cleanup()
+
```
Comment (mheon) [inline test/e2e/exec_test.go]: Nit: empty line

File: test/e2e/exec_test.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	})
+
+	It("podman rmi bogus image", func() {
+		session := podmanTest.Podman([]string{"rmi", "debian:6.0.10"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(125))
+
```
Comment (mheon) [inline test/e2e/exec_test.go]: Nit: empty line

**Comment by mheon** [comment]
Bunch of extra whitespace, but otherwise LGTM. Faster we deprecate BATS the better

**Comment by baude** [comment]
cleaned up the whitespaces

File: test/e2e/diff_test.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
+package integration
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Podman rmi", func() {
```
Comment (TomSweeneyRedHat) [inline test/e2e/diff_test.go]: The file name is diff_test but this looks to be all rmi tests?

File: test/e2e/exec_test.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
+package integration
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Podman rmi", func() {
```
Comment (TomSweeneyRedHat) [inline test/e2e/exec_test.go]: Ditto this, this is the exec test but it looks like all rmi again?  Do I need more tea?

File: test/e2e/history_test.go [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"path/filepath"
+)
+
+var _ = Describe("Podman export", func() {
```
Comment (TomSweeneyRedHat) [inline test/e2e/history_test.go]: And this history_test looks to be the export test repeated?

File: test/e2e/diff_test.go (hunk 7 above)
Comment (baude) [reply test/e2e/diff_test.go]: fixed

File: test/e2e/exec_test.go (hunk 8 above)
Comment (baude) [reply test/e2e/exec_test.go]: fixed

File: test/e2e/history_test.go (hunk 9 above)
Comment (baude) [reply test/e2e/history_test.go]: fixed

File: test/e2e/history_test.go [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+		podmanTest = PodmanCreate(tempdir)
+		podmanTest.RestoreAllArtifacts()
+	})
+
+	AfterEach(func() {
+		podmanTest.Cleanup()
+
+	})
+
+	It("podman export output flag", func() {
```
Comment (mheon) [inline test/e2e/history_test.go]: This looks like an export test?

**Comment by baude** [comment]
bot, retest this please

**Comment by mheon** [comment]
@baude I think some of your tests are still in the wrong files

**Comment by baude** [comment]
Yeah it looks like some bad git action happened ... don't merge until I get a chance to fix this please.

File: Makefile [hunk 11]
```diff
@@ -194,7 +194,10 @@ install.tools: .install.gitvalidation .install.gometalinter .install.md2man
 
 .install.md2man: .gopathok
 	if [ ! -x "$(GOPATH)/bin/go-md2man" ]; then \
-		go get -u github.com/cpuguy83/go-md2man; \
+		   go get github.com/cpuguy83/go-md2man; \
```
Comment (mheon) [inline Makefile]: Use `-d` flag here so we stop after downloading and don't install the bad version

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**